eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -186,27 +184,31 @@ object Helpers {
-   * then we use the real scid.
-   *
-   * Private channels are handled like public channels that have not yet been announced, there is no special case.
-   *
-   * Decision tree:
-   *  - received remote_alias from peer
-   *    - before channel announcement: use remote_alias
-   *    - after channel announcement: use real scid
-   *  - no remote_alias from peer
-   *    - min_depth > 0: use real scid (may change if reorg between min_depth and 6 conf)
-   *    - min_depth = 0 (zero-conf): spec violation, our peer MUST send an alias when using zero-conf
+   * We use our local alias for our channel_update until the channel is publicly announced; then we use the real scid.
    */
-  def scidForChannelUpdate(channelAnnouncement_opt: Option[ChannelAnnouncement], shortIds: ShortIds): ShortChannelId = {
-    channelAnnouncement_opt.map(_.shortChannelId) // we use the real "final" scid when it is publicly announced
-      .orElse(shortIds.remoteAlias_opt) // otherwise the remote alias
-      .orElse(shortIds.real.toOption) // if we don't have a remote alias, we use the real scid (which could change because the funding tx possibly has less than 6 confs here)
-      .getOrElse(throw new RuntimeException("this is a zero-conf channel and no alias was provided in channel_ready")) // if we don't have a real scid, it means this is a zero-conf channel and our peer must have sent an alias
+  def scidForChannelUpdate(channelAnnouncement_opt: Option[ChannelAnnouncement], shortIds: ShortIds, remoteFeatures: Features[InitFeature]): ShortChannelId = {
+    channelAnnouncement_opt match {
+      case Some(ann) => ann.shortChannelId // we use the real "final" scid when it is publicly announced
+      case None if remoteFeatures.hasFeature(Features.ScidAlias) => shortIds.localAlias // otherwise we use our local alias if our peer supports it
+      case None => shortIds.real.toOption match {
+        case Some(real) => real // we use the real scid if our peer doesn't support scid_alias and the channel is confirmed
+        case None => shortIds.localAlias // spec violation: this is a 0-conf channel, our peer MUST support scid_alias
+      }
+    }
   }
 
-  def scidForChannelUpdate(d: DATA_NORMAL): ShortChannelId = scidForChannelUpdate(d.channelAnnouncement, d.shortIds)
+  def scidForChannelUpdate(d: DATA_NORMAL): ShortChannelId = scidForChannelUpdate(d.channelAnnouncement, d.shortIds, d.commitments.remoteParams.initFeatures)
+
+  /**
+   * If our peer sent us an alias, that's what we must use in the channel_update we send them to ensure they're able to
+   * match this update with the corresponding local channel.
+   */
+  def channelUpdateForDirectPeer(nodeParams: NodeParams, channelUpdate: ChannelUpdate, remoteAlias_opt: Option[Alias]): ChannelUpdate = {
+    remoteAlias_opt match {
+      case Some(remoteAlias) => Announcements.signChannelUpdate(nodeParams.privateKey, channelUpdate.copy(shortChannelId = remoteAlias))
+      case None => channelUpdate
```

Comment:
This is, in fact, where the spec violation would appear, if `!nodeParams.remoteParams.features.hasFeature(Features.ScidAlias) && remoteAlias_opt.isEmpty`

```suggestion
      case None => channelUpdate // note that if this is a zero-conf channel and the peer hasn't provided an alias, then the channel_update will have our local alias; but it's harmless to us, and they violated the spec, so it's their problem.
```